src/ComponentInstaller.php

```diff
@@ -187,12 +191,89 @@ public function onPostPackageInstall(PackageEvent $event)
                 return $injectors;
             }, new Collection([]))
             // Inject modules into configuration
-            ->each(function ($injector, $module) use ($name, $packageTypes) {
+            ->each(function ($injector, $module) use ($name, $packageTypes, $applicationModules) {
+                $injector->setApplicationModules($applicationModules);
                 $this->injectModuleIntoConfig($name, $module, $injector, $packageTypes);
             });
     }
 
     /**
+     * Find all Module classes in the package and include them.
+     * Module classes is used later
+     * @see \Zend\ComponentInstaller\Injector\AbstractInjector::injectAfterDependencies
+     * - to get package dependencies - module method `getModuleDependencies`
+     * and add component in a correct order on the module list.
+     *
+     * It works currently with PSR-0 and PSR-4 autoloaded packages.
+     *
+     * @param PackageInterface $package
+     * @return void
+     */
+    private function includeModuleClasses(PackageInterface $package)
+    {
+        $installer = $this->composer->getInstallationManager();
+        $packagePath = $installer->getInstallPath($package);
+
+        $autoload = $package->getAutoload();
+        foreach ($autoload as $type => $map) {
+            foreach ($map as $namespace => $path) {
+                switch ($type) {
```

Comment:
Done in b45896e8f61c8225e0484a6e4a94d49e434c7f2a.

I've added support for `classmap` and `files` autoloading.
BTW. I think in Apigility it was `classmap` before ;)